bot/discordbot.py

```diff
@@ -174,6 +178,99 @@ async def hello(interaction: discord.Interaction):
+        return
+    
+    # Check if the server already has a valid credential
+    server_id = interaction.guild_id
+    BACKEND_URL = env["DISCORD_BACKEND_URL"]
+    response = requests.get(
+        BACKEND_URL + '/api/?serverId=' + str(server_id)
+    )
+    if response.status_code != 400:
+        await interaction.response.send_message(
+            "The server already has a valid credential.",
+            ephemeral=True
+        )
+        return
+    
+    # Check if the user is a server owner or server admin, and 
+    author = interaction.user
+    if not author.id == interaction.guild.owner_id:
+        is_admin = False
+        for role in author.roles:
+            if role.permissions.administrator:
+                is_admin = True
+                break
+        if not is_admin:
+            # the user is not authorized as an admin role
+            await interaction.response.send_message(
+                content="You are not authorized to register, reach out to the server admins to do so.",
+                ephemeral=True
+            )
+            return
+        
+    # Check if email is valid
+    email_pattern = re.compile(r'^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+    if not re.match(email_pattern, email):
+        await interaction.response.send_message("Invalid email address format. Please provide a valid email address.")
+        return
+    
+    registration_url ="{0}/payment/register/".format(dc_url)
+    
+    with requests.Session() as s:
+        try:
+            response = s.get(registration_url)
+            response.raise_for_status()
+        except:
+            await interaction.response.send_message("An error occurred while processing your request. Please try agian later or contact with humanID.")
+
+        soup = BeautifulSoup(response.content, 'html.parser') # Parse the HTML content
+        csrf_token_input = soup.find('input', {'type': 'hidden', 'name': 'csrfmiddlewaretoken'})
+        if csrf_token_input:
+            csrf_token = csrf_token_input.get('value')
+            form_data = {
+                'first_name': 'first',
+                'last_name': 'last',
+                'email': email,
+                'password': str(DCPASSWORD), 
+                'passConf': str(DCPASSWORD) 
+            }
+            headers = {'X-CSRFToken': csrf_token,
+                       'X-Discord-Bot': str(server_id)}
+            try:
+                # Make a POST request to the RegisterUser function
+                response = s.post(registration_url, data=form_data, headers=headers)
+                response.raise_for_status()  
+                dup_message = 'This email is already in use. Login instead?'   #message that imply a duplicate email
+                if dup_message in response.text:
+                    await interaction.response.send_message("This email is already in use or needs to be activated. Please use another one or activate it use the activation link sent to this email.")
```

Comment:
Found another type "...activate is ~use~ the activation link sent to this email

Reply:
should be "using" instead